messages_test.go

```diff
@@ -1,39 +1,250 @@
 package main
 
-import "testing"
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+	"time"
 
-func TestString(t *testing.T) {
-	type test struct {
-		Log      Log
-		expected string
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const (
+	TestAuthorName  = "John Smith"
+	TestAuthorEmail = "john.smith@example.com"
+)
+
+func Test_New(t *testing.T) {
+	Convey("When creating a log with New", t, func() {
+		rc = mockClock{}
+
+		cfg := &Config{
+			Author: TestAuthorName,
+			Email:  TestAuthorEmail,
+		}
+
+		Convey("Should return a new log", func() {
+			log := New("New Log", cfg)
+
+			So(log.Author, ShouldEqual, TestAuthorName)
+		})
+	})
+}
+
+func Test_SaveNewLog(t *testing.T) {
+	Convey("When saving a log file", t, func() {
+		err := SaveNewLog("beacon_test.json", []Log{})
+		So(err, ShouldEqual, nil)
+	})
+}
+
+func Test_ShowLog(t *testing.T) {
+	cfg := &Config{
+		Author: TestAuthorName,
+		Email:  TestAuthorEmail,
 	}
+	Convey("When ShowLog is called", t, func() {
+		logs := []Log{
+			New("First", cfg),
+			New("Second", cfg),
+		}
+
+		Convey("With -1 for the count", func() {
+			out = &mockWriter{}
 
-	testArray := []struct {
-		Log      Log
-		expected string
-	}{
-		{
-			Log{
-				Date:    1502569641,
-				Email:   "j@jh.com",
-				Author:  "John Henry",
-				Message: "I broke lots of things!",
-			},
-			`
+			Convey("Should return all logs", func() {
+				ShowLog(logs, -1)
+				w := out.(*mockWriter)
+				So(w.String(), ShouldEqual, `
+==========================================
```

Comment:
do you think our tests will become brittle by having to, by hand, copy and paste these kinds of things? 

For example the usage of : 
`============================`
seems like a pain to have to copy and paste the exact amount `=` to have the test succeed.

I mean, sort of related, we'll probably be removin those in favour of just seperating the log entries printed out by `\n`.